test/index.js

```diff
@@ -1892,8 +1892,62 @@ describe('Nightmare', function () {
       nightmare = Nightmare({ electronPath: require('electron') });
       nightmare.should.be.ok;
     });
+
+    it('should allow to use external Promise', function*() {
+      nightmare = Nightmare({ Promise: require('bluebird') });
```

Comment:
The only reason I didn't do that is because people like myself who regularly use external Promise libs usually assign them to the `Promise` var in the beginning of their scripts. Doing so here overrides the global promise lib not just in the context of this function/test, but also globally (b/c `Promise` is defined globally w/ `var` & not `let`) so that will mess up other tests.

The other option is doing something like this, but I'm not a fan of how it looks by-comparison lol:
```js
const Promise2 = require('bluebird');
nightmare = Nightmare({ Promise: Promise2 });
```